modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2846,6 +2849,26 @@ private void _validateOneToOneUpdate(
 		}
 	}
 
+	private Map<String, Serializable> _validatePicklistDefaultValue(
+		DynamicObjectDefinitionTable dynamicObjectDefinitionTable,
+		Map<String, Serializable> values) {
+
+		List<ObjectField> objectFields =
+			dynamicObjectDefinitionTable.getObjectFields();
+
+		for (ObjectField objectField : objectFields) {
+			if (!values.containsKey(objectField.getName()) &&
+				Objects.equals(objectField.getBusinessType(), "Picklist") &&
+				Validator.isNotNull(objectField.getDefaultValue())) {
```

Comment:
Hi @pedro-oliveira446, we don't need to check `Validator.isNotNull(objectField.getDefaultValue())` because when the business type of object field is "Picklist" the default value is mandatory. So it will never be null.

In addition, we can use the `ObjectFieldConstants.BUSINESS_TYPE_PICKLIST` to get the "Picklist" value instead of writing it straight in the code.

Lastly, we can check the business type before checking if the object's field name contains in the value map.